File: lib/services/food_details_client.dart
import '../models/food_details.dart';
import '../models/ingredient.dart';
import 'open_food_facts_service.dart';

abstract class FoodDetailsClient {
  Future<FoodDetails?> lookup(Ingredient ingredient);
}

class OpenFoodFactsDetailsClient implements FoodDetailsClient {
  const OpenFoodFactsDetailsClient();

  @override
  Future<FoodDetails?> lookup(Ingredient ingredient) {
    return OpenFoodFactsService.lookupDetails(
      name: ingredient.name,
      barcode: ingredient.barcode,
    );
  }
}

File: apps/ios/FreshPantry/Services/FoodDetailsClient.swift
import Foundation

/// Food-details lookup backed by `OpenFoodFactsService` — barcode-first, then
/// name search. `Sendable` so it can cross the actor / store boundary; stateless.
/// The OFF service is best-effort (errors → nil internally), so `lookup` never
/// throws in practice but keeps `throws` for an alternate backend.
struct OpenFoodFactsDetailsClient: Sendable {
    func lookup(_ ingredient: Ingredient) async throws -> FoodDetails? {
        await OpenFoodFactsService.lookupDetails(
            name: ingredient.name,
            barcode: ingredient.barcode
        )
    }
}
